From 2663cb97aae48d23ae22c232830ceed885b4911e Mon Sep 17 00:00:00 2001
From: "Yang, Dong" <dong.yang@intel.com>
Date: Tue, 2 Nov 2021 08:11:27 +0800
Subject: [PATCH] WA camera rotation

based on WA patch1, change camera rotation from 270 to 180 for classin
APP

Tracked-On:
Signed-off-by: Yang, Dong <dong.yang@intel.com>
---
 core/jni/android_hardware_Camera.cpp | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/core/jni/android_hardware_Camera.cpp b/core/jni/android_hardware_Camera.cpp
index d46c77396960..eee72700a29f 100644
--- a/core/jni/android_hardware_Camera.cpp
+++ b/core/jni/android_hardware_Camera.cpp
@@ -545,8 +545,14 @@ static void android_hardware_Camera_getCameraInfo(JNIEnv *env, jobject thiz,
         return;
     }
     if(checkflag == 1){
-       if(cameraInfo.facing == 1) 
-        env->SetIntField(info_obj, fields.orientation, 270);
+        if(cameraId == 0) {
+            cameraInfo.facing = 0;
+            env->SetIntField(info_obj, fields.orientation, 0);
+        }
+        else{
+            cameraInfo.facing = 1;
+            env->SetIntField(info_obj, fields.orientation, 180);
+        }
     }
     else{
         env->SetIntField(info_obj, fields.orientation, cameraInfo.orientation);
@@ -631,7 +637,7 @@ static jint android_hardware_Camera_native_setup(JNIEnv *env, jobject thiz,
     }
     if(checkflag == 1){
       if(cameraInfo.facing == CAMERA_FACING_FRONT)
-         cameraInfo.orientation = 270;
+         cameraInfo.orientation = 180;
     } 
     else 
        cameraInfo.orientation = 0;
-- 
2.27.0

